test/integ-new/main_new_test.go

```diff
@@ -401,3 +434,44 @@ func testValidateKeyVersionState(ctx context.Context, tb testing.TB, kmsClient *
 		tb.Errorf("primary was set to version %d, but expected %d", primaryNumber, expectedPrimary)
 	}
 }
+
+// This is intended to mock an event where we need to emergently rotate the key.
+// We disable the key version and remove it as primary.
+func testEmergentDisable(ctx context.Context, tb testing.TB, kmsClient *kms.KeyManagementClient, keyName, versionName string) {
+	tb.Helper()
+	ver, err := kmsClient.GetCryptoKeyVersion(ctx, &kmspb.GetCryptoKeyVersionRequest{Name: versionName})
+	if err != nil {
+		tb.Fatalf("unable to retrieve version %s: %s", versionName, err)
+	}
+
+	ver.State = kmspb.CryptoKeyVersion_DISABLED
+	var messageType *kmspb.CryptoKeyVersion
+	mask, err := fieldmaskpb.New(messageType, "state")
+	if err != nil {
+		tb.Fatalf("unable to create field mask: %s", err)
+	}
+	updateReq := &kmspb.UpdateCryptoKeyVersionRequest{
+		CryptoKeyVersion: ver,
+		UpdateMask:       mask,
+	}
+	if _, err := kmsClient.UpdateCryptoKeyVersion(ctx, updateReq); err != nil {
+		tb.Fatalf("unable to disable version: %s", err)
+	}
+
+	key, err := kmsClient.GetCryptoKey(ctx, &kmspb.GetCryptoKeyRequest{Name: keyName})
+	if err != nil {
+		tb.Fatalf("unable to retrieve key %s: %s", keyName, err)
+	}
+
+	labels := make(map[string]string, 0)
+	key.Labels = labels
+
+	var mT *kmspb.CryptoKey
+	mask, err = fieldmaskpb.New(mT, "labels")
+	if err != nil {
+		tb.Fatalf("unable to create field mask: %s", err)
+	}
+	if _, err := kmsClient.UpdateCryptoKey(ctx, &kmspb.UpdateCryptoKeyRequest{CryptoKey: key, UpdateMask: mask}); err != nil {
+		tb.Fatalf("unable to set labels: %s", err)
+	}
```

Comment:
You can set the labels and state in the same RPC. Just add both fields and use both in the update mask:

```go
        req := &kmspb.UpdateCryptoKeyVersionRequest{
                CryptoKeyVersion: &kmspb.CryptoKeyVersion{
                        Name:  name,
                        State: kmspb.CryptoKeyVersion_DISABLED,
                        Labels: labels,
                },
                UpdateMask: &fieldmask.FieldMask{
                        Paths: []string{"state", "labels"},
                },
        }
```

Reply:
I think `state` is for a key version, and `labels` is for a key. [CryptoKey]https://pkg.go.dev/cloud.google.com/go/kms@v1.15.3/apiv1/kmspb#CryptoKey doesn't have a `state` field and [CryptoKeyVersion](https://pkg.go.dev/cloud.google.com/go/kms@v1.15.3/apiv1/kmspb#CryptoKeyVersion) doesn't have a `labels` field.

Reply:
Ah you're right, sorry. What labels are we removing here - just `primary`? This is deleting all labels. Also, it might be better to remove the lable and _then_ delete the key. Otherwise you have a race condition where the label is pointing to a disabled key.

Reply:
Thanks for pointing out the race condition, I have update the code.